lib/optimize/InnerGraphPlugin.js

```diff
@@ -116,27 +116,27 @@ class InnerGraphPlugin {
 					});
 
 					parser.hooks.finish.tap("InnerGraphPlugin", () => {
-						const innerGraphState = InnerGraph.getState(parser.state);
-						if (!innerGraphState) return;
+						if (!InnerGraph.isEnabled(parser.state)) return;
 
-						const {
-							innerGraph,
-							allExportDependentDependencies
-						} = innerGraphState;
+						const allExportDependentDependencies = InnerGraph.getExportDependentDependencies(
+							parser.state
+						);
 						// flatten graph to terminal nodes (string, undefined or true)
```

Comment:
move the whole algorithm into InnerGraph

Reply:
`InnerGraph.inferDependencyUsage(): Map<Dependency, Set<string | TopLevelSymbol> | true>`

This should make it possible to
* remove `getNodesIterable`
* replace `getExportDependentDependencies` with `InnerGraph.addDependency`